<launch>
    <arg name="config_path" default="$(find bw_tags)/config/tags.yaml" />
    <arg name="image_topic" default="image_raw" />
    <arg name="info_topic" default="camera_info" />

    <group ns="bw">
        <node pkg="bw_tags" type="tag_marker_publisher.py" name="tag_marker_publisher" required="false" output="screen">
            <param name="base_frame" value="base_link"/>
        </node>
        <node pkg="bw_tags" type="tag_request_server" name="tag_request_server" required="false" output="screen">
            <remap from="in/image_raw" to="$(arg image_topic)" />
            <remap from="in/camera_info" to="$(arg info_topic)" />
            <remap from="out/image_raw" to="tag_image" />
            <remap from="out/camera_info" to="tag_camera_info" />
        </node>

        <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag" required="false" output="screen" >
            <rosparam command="load" file="$(arg config_path)" ns="" />

            <remap from="camera_info" to="tag_camera_info" />
            <remap from="image_rect" to="tag_image" />

            <param name="remove_duplicates" type="bool" value="true" />
            <param name="publish_tag_detections_image" type="bool" value="true" />
        </node>
    </group>
</launch>
